bin/parse_podfile_lock

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
+
+require 'json'
+
+dependency_metadata_regex = /-\s(?<pod>.*?)\s\((=|(~>))?\s?(?<version>(\d*\.?)+)\)/
```

Comment:
I'm worried about this regex.  Does cocoa provide dependencies in structured formats like json?

Reply:
Not that I know of. There's a `pod list` command that's supposed to list pods but I'm not able to get it to work locally with `Podfile.lock`s. There was also an individually maintained tool called [`cocoapods-dependencies`](https://github.com/segiddins/cocoapods-dependencies), but it's gone three years without an update now, so I'm not sure we should use it. I'm open to thoughts, though!

Reply:
Is the lock file itself always a yaml? It looks like so from the lockfiles you added to specs.  If so, you can just call yaml parse to get a structured objected of the values.

Reply:
Actually, please take a look at the `to_yaml` method here
https://www.rubydoc.info/github/CocoaPods/Core/Pod/Lockfile#to_yaml-instance_method

Reply:
Thanks for pointing that out, I didn't notice it's valid YAML. It's now using YAML to parse the dependency list, but because the version of each dependency is in the same string as the name it's still using a (smaller) regex.

Difference is before the regex grabbed every dependency and version of the file. Now the file is parsed as YAML and regex is used only to extract the name and version from each dependency element string.